src/runner/settings_window.cpp

```diff
@@ -132,14 +132,17 @@ DWORD g_settings_process_id = 0;
 
 void run_settings_window()
 {
-    STARTUPINFO startup_info = { sizeof(startup_info) };
     PROCESS_INFORMATION process_info = { 0 };
-    HANDLE process = NULL;
-    HANDLE hToken = NULL;
-    STARTUPINFOEX siex = { 0 };
-    PPROC_THREAD_ATTRIBUTE_LIST pptal = NULL;
+    HANDLE process = nullptr;
+    HANDLE hToken = nullptr;
+    PPROC_THREAD_ATTRIBUTE_LIST pptal = nullptr;
     WCHAR executable_path[MAX_PATH];
-    GetModuleFileName(NULL, executable_path, MAX_PATH);
+
+    if (0 == GetModuleFileName(nullptr, executable_path, MAX_PATH))
```

Comment:
That reminds me of a similar discussion about `MAX_PATH` [here](https://github.com/microsoft/PowerToys/pull/884#discussion_r355341906). 
[The docs](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file) do not list `GetModuleFileName` as a function with a path length restrictions lifted if you opt-in to long path behavior, but [`GetModuleFileName` docs](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamew) state that it's always possible to get a long path from it:
> `Therefore, the path can be a long or short file name, and can use the prefix "\?"`

Maybe we should write a wrapper for the function for people to use, since it seems like a common op to do, e.g.:
```cpp
std::wstring get_module_filename(HMODULE mod = nullptr)
{
  wchar_t buffer[MAX_PATH + 1];
  DWORD actual_length = GetModuleFileNameW(mod, buffer, MAX_PATH);
  if(GetLastError() == ERROR_INSUFFICIENT_BUFFER)
  {
    const DWORD long_path_length = 0xFFFF; // should be always enough
    std::wstring long_filename(long_path_length, L'\0');
    actual_length = GetModuleFileNameW(mod, long_filename.data(), long_path_length);
    return long_filename.substr(0, actual_length);
  }
  return {buffer, actual_length};
}
```